src/citrine/exceptions.py

```diff
@@ -33,27 +33,48 @@ class NonRetryableHttpException(NonRetryableException):
     """An exception originating from an HTTP error from a Citrine API."""
 
     def __init__(self, path: str, response: Optional[Response] = None):
-        super().__init__(path)
         self.url = path
+        self.detailed_error_info = []
         if response is not None:
             self.response_text = response.text
             self.code = response.status_code
+
+            method = "unknown"
+            if response.request is not None:
+                method = response.request.method
+
+            self.detailed_error_info = ["{} (code: {}) returned from {} request to "
+                                        "path: '{}'".format(response.reason, self.code,
+                                                            method, path)]
```

Comment:
Functionally equivalent, but future-proofs a bit
```suggestion
            self.detailed_error_info.append(
                "{} (code: {}) returned from {} request to path: '{}'".format(
                    response.reason, self.code, method, path
                )
            )
```